### PR TITLE
DOC: Add .to_xyz to io docs

### LINKED ISSUE #41839: DOC: Input/Output page contains only input functions
#### Location of the documentation

https://pandas.pydata.org/pandas-docs/dev/reference/io.html#html

#### Documentation problem

The title is Input/Output but all linked methods are `read_xyz()`, and there is little if any reference to any `write_xyz()` or `to_xyzz methods.

#### Suggested fix for documentation

Perform a review and enure all input/output methods are referenced on the page.

### PR DESCRIPTION
**Pull request by gansel51**
- [x] closes #41839
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

Screenshot below for updated docs

Screen Shot 2021-06-06 at 7 03 23 PM

### CODE REVIEW

File: doc/source/reference/io.rst [hunk 1]
```diff
@@ -29,31 +31,34 @@ Clipboard
...
    read_clipboard
+   DataFrame.to_clipboard
 
 Excel
 ~~~~~
 .. autosummary::
    :toctree: api/
 
    read_excel
+   DataFrame.to_excel
```
Comment (attack68) [inline doc/source/reference/io.rst:42]: can you also add `Styler.to_excel`

File: doc/source/reference/io.rst [hunk 2]
```diff
@@ -67,13 +72,15 @@ HTML
    :toctree: api/
 
    read_html
+   DataFrame.to_html
```
Comment (attack68) [inline doc/source/reference/io.rst:75]: Can you also add as a separate line `Styler.to_html`

**Review by attack68 — CHANGES_REQUESTED** [review]
Thanks v m @gansel51. 

Few comments in line and also:

Can you add a section on `LaTeX` and add:
`DataFrame.to_latex` and `Styler.to_latex`

**Comment by gansel51** [comment]
@attack68 For sure. I added the `DataFrame.to_latex` method on a feature branch I'll merge when complete. Looking into the Styler methods now because an import error blocked my original request.

**Comment by gansel51** [comment]
@attack68 Added those methods, ready for re-review!

**Comment by gansel51** [comment]
Screen Shot 2021-06-07 at 1 15 28 PM

@attack68 yeah, here's a screenshot of the LaTeX section which shows a cross section of the three types of changes:
1. Add new section
2. Add DataFrame.to_xyz
3. Add Syle.to_xyz

**Review by attack68 — APPROVED** [review]
LGTM.

This is better with the increased info and links. Would be nice if as a follow on we could get the columns aligning, i.e. by merging the Styler and DataFrame tocs so they dont have different alignment.

**Review by WillAyd — COMMENTED** [review]
Thanks for the PR. Just out of curiosity why are some but not all of these namespaced with `DataFrame...`

**Comment by gansel51** [comment]
@WillAyd The `DataFrame.to_xyz` methods have the DataFrame modifier because they are DataFrame methods and must be done on one. The remaining IO methods aren't done on DataFrames (just `pandas` or `Styler`) so didn't have that prefix. It both lets Sphynx pick up the methods (otherwise they won't be found by the autosummary) and gives users more accurate usage information.

**Review by attack68 — CHANGES_REQUESTED** [review]
There seems to be some CI errors. The docs don't build without errors, I think because the current module is changed in one part of `pandas.io.formats.style` and then `ExcelWriter` is referenced.

**Comment by gansel51** [comment]
> There seems to be some CI errors. The docs don't build without errors, I think because the current module is changed in one part of `pandas.io.formats.style` and then `ExcelWriter` is referenced.

The docs didn't build without errors when I first tried pre-PR either, but I switched the current module back for ExcelWriter and it is generating properly. See screenshot below for ExcelWriter

Screen Shot 2021-06-08 at 11 14 55 AM

**Comment by gansel51** [comment]
@attack68 I went ahead and resolved the warnings on the page as well. The page is rendering properly for me with no errors.

**Comment by jreback** [comment]
lgtm. can merge on green.

**Comment by gansel51** [comment]
> lgtm. can merge on green.

Thanks @jreback will keep an eye on the checks

**Comment by gansel51** [comment]
> https://github.com/pandas-dev/pandas/pull/41844/checks?check_run_id=2777904153

I can switch that, but `LaTeX` is the proper capitalization for the project (see [LaTeX Website](https://www.latex-project.org/)). Please advise if I should switch it or if there is a way to override that check @jreback

**Comment by jreback** [comment]
actually this is coming from our validate_rst_ checker cc @MarcoGorelli i think there is a way to ignore it

**Comment by gansel51** [comment]
@jreback cc @MarcoGorelli Any update here?

**Comment by simonjayhawkins** [comment]
moving off 1.3 milestone for now.

**Comment by MarcoGorelli** [comment]
Looks like the error is
```
/home/runner/work/pandas/pandas/pandas/io/json/_normalize.py:240:SS04:pandas.io.json.json_normalize:Summary contains heading whitespaces
```

**Comment by gansel51** [comment]
@jreback @MarcoGorelli @attack68 I removed the deprecated method from the documentation, kicking off a new build could (should?) solve the issue so this PR can merge. I can definitely look into putting LaTeX in the allowlisted next!

**Comment by gansel51** [comment]
@attack68 I'm not seeing where the checks are failing and not just failing because of a cancellation, can you point me in the right direction?

**Comment by simonjayhawkins** [comment]
```
/home/runner/work/pandas/pandas/doc/source/reference/api/pandas.to_pickle.rst:40: WARNING: Citation [Recbd1b9f7896-1] is not referenced.
```
in https://github.com/pandas-dev/pandas/pull/41844/checks?check_run_id=2800940727

**Comment by jreback** [comment]
@gansel51 this is ok for 1.3 after its passing.

**Comment by attack68** [comment]
> @attack68 I'm not seeing where the checks are failing and not just failing because of a cancellation, can you point me in the right direction?

isnt this because it should be `DataFrame.to_pickle`. if `pandas.to_pickle` isnt there that will produce the error simonjayhawkins highlighted

**Comment by gansel51** [comment]
@attack68 @simonjayhawkins Thanks for pointing it out, I was having trouble isolating the error from the logs. Made that change, will keep an eye on the workflow status.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @gansel51

**Comment by jreback** [comment]
@meeseeksdev backport 1.3.x

**Pull request merged**